### PR TITLE
09-coap: port to pytest-based framework

### PR DESCRIPTION
**Pull request by miri64**
This provides automated tests for the new, pytest-based framework for spec09.

They are still a bit convoluted, I think, but maybe @chrysn can help here as our local `aiocoap`/`asyncio` expert ^^. It bugs me especially, that for `task02` I had to introduce a separate script that is started externally, but otherwise I ran into endless loops or too early canceled processes.

I kept the old scripts, since the specs are explicitly referring to them.

Requires #155

### CODE REVIEW

**Comment by miri64** [comment]
Squashed and rebased to current master

**Comment by miri64** [comment]
Rebased to current master

**Comment by miri64** [comment]
> Rebased to current master

Or not..

**Comment by miri64** [comment]
Rebased to current master.

**Comment by fjmolinas** [comment]
Can you remove the old scripts as well?

**Comment by miri64** [comment]
> Can you remove the old scripts as well?

No, see OP:

> I kept the old scripts, since the specs are explicitly referring to them.

**Comment by fjmolinas** [comment]
BTW I tagged @kb2ma just in the case he might have some input on these tests since I know he did a lot of work with automatizing coap testing on his side and that he was using pytest as well, (although he is not actively working on it anymore). Here is the reference to this past work https://github.com/kb2ma/riot-coap-pytest.

**Comment by fjmolinas** [comment]
> No, see OP:

 Ah ok, sorry I jumped directly to the code (top of the list for things you shouldn't do, first read docs!)

**Comment by miri64** [comment]
@chrysn do you maybe want to take a look?

**Comment by fjmolinas** [comment]
> They are still a bit convoluted, I think, but maybe @chrysn can help here as our local aiocoap/asyncio expert ^^.

ping @chrysn :)

**Comment by miri64** [comment]
And another rebase (should be the last one as this is the only remaining test spec)

**Comment by miri64** [comment]
I was able to remove the separate `server.py` script.

File: 09-coap/test_spec09.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+NODE_ULA = "fd00:bbbb::2"
+TAP = 'tap0'
+
+
+pytestmark = pytest.mark.rc_only()
+
+
+# pylint: disable=R0903
+class CordEPRegisterParser(ShellInteractionParser):
+    # TODO: provide upstream
```
[the remark this answers is not in the document]
Comment (fjmolinas) [reply 09-coap/test_spec09.py:33]: Ah there is this remaining still, you can still merge and address @miri64

[the remark this answers is not in the document]
Comment (miri64) [reply 09-coap/test_spec09.py:33]: See https://github.com/RIOT-OS/RIOT/issues/14650

**Pull request merged**